### PR TITLE
Add styles matching Moodle defaults for Moodle pages

### PR DESCRIPTION
**Pull request by acelaya**
Following what was done in https://github.com/hypothesis/lms/pull/5761 for Canvas Pages, this PR defines some basic styles to make pages loaded in assignments look as close as possible to Moodle Pages loaded inside Moodle itself.

Moodle seems to have a single `all` stylesheet that's apparently the server-side merging of a bunch of stylesheets and served in a single `/theme/styles.php/boost/{timestamp}/all` URL.

I observed pages do not add any classes to the generated markup, but only a few inlined styles. Based on that I took the bare minimum from that `all` stylesheet so that it looks close enough, which in essence only includes styles applied directly to tags.

### Testing steps

1. Check out this branch.
2. If your LMS server is already running, run `yarn gulp build-css`
3. Open this Moodle page: https://hypothesisuniversity.moodlecloud.com/mod/page/view.php?id=860&forceview=1
4. Open this Moodle assignment, which uses that page: https://hypothesisuniversity.moodlecloud.com/mod/lti/view.php?id=865
5. Check that styles are close enough.

### TODO

- [x] Simplify styles as much as possible
- [x] Find out if styles in Moodle instances can be customized (colors and such) -> I could not find this anyway in settings and such. That doesn't mean this is not possible in self-hosted Moodle instances.

### CODE REVIEW

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+sub {
+  bottom: -0.25em;
+}
+
+sup {
+  top: -0.5em;
+}
+
+a {
+  color: #ec7f13;
```
Comment (acelaya) [inline lms/static/styles/moodle_pages/moodle_pages.scss:78]: This is Moodle's brand color. I'm not sure if this color can be change in an instance and therefore may need to be dynamic.

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 2]
```diff
@@ -0,0 +1,226 @@
+body {
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
+    'Helvetica Neue', Arial, 'Noto Sans', 'Liberation Sans', sans-serif,
+    'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
+  font-size: 0.9375rem;
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:5]: This value looks like something that was produced by a SASS compiler or something similar. Have you got a link for the original source of truth in the Moodle source repository?

Comment (acelaya) [reply lms/static/styles/moodle_pages/moodle_pages.scss:5]: Good point. Let me check that.

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 3]
```diff
@@ -0,0 +1,226 @@
+body {
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:1]: The `canvas_pages.scss` module for Canvas has some comments at the top has some notes about where the styles came from and upstream links that are useful if we need to update it in future. I think it would be useful to have  something similar here. In the case of fonts, it looks like Moodle is relying on system fonts, so we don't need any imports for those.

File: lms/static/styles/moodle_pages/moodle_pages.scss (hunk 2 above)
Comment (acelaya) [reply lms/static/styles/moodle_pages/moodle_pages.scss]: It's a hardcoded value. In some changelog they mention the use of this value after updating to bootstrap 4, as it translates to ~15px in most browsers.

https://github.com/moodle/moodle/blob/38a3310c9272edfbac77ee4a8527e7838642072e/theme/boost/upgrade.txt#L93

File: lms/static/styles/moodle_pages/moodle_pages.scss (hunk 3 above)
Comment (acelaya) [reply lms/static/styles/moodle_pages/moodle_pages.scss:15]: I have added a comment to the top, referencing the main files which should be checked in case new updates are needed.

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+//   * https://github.com/moodle/moodle/blob/main/theme/boost/scss/bootstrap/_variables.scss
+
+body {
+  // Defined as $font-family-sans-serif
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
+    'Helvetica Neue', Arial, 'Noto Sans', 'Liberation Sans', sans-serif,
+    'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
+  font-size: 0.9375rem; // Defined as `$font-size-base`
+  line-height: 1.5;
+  color: #1d2125; // Defined as `$gray-900`
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:22]: This is a SASS file, so you might as well define matching variables rather than write out comments to the same effect. I see that `$font-size-base` really is defined as `0.9375rem` upstream. Assuming a base size of 16px, that equals 15px.

**Review by marcospri — APPROVED** [review]
For my limited understanding of this, looks good :+1:

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 5]
```diff
@@ -0,0 +1,249 @@
+// This stylesheet contains a re-implementation of some of Moodle styles for
+// pages, so that Moodle pages served through our LMS app + Via look close to
+// how the page looks in Moodle.
+//
+// The markup generated for pages does not include specific classes, so we only
+// need to define styles for HTML elements.
+//
+// The styles defined here are mainly extracted from the next Moodle files:
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:8]:
```suggestion
// The styles defined here are mainly extracted from the following Moodle files:
```

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+  margin-left: 0;
+}
+
+blockquote {
+  margin: 0 0 1rem;
+}
+
+b,
+strong {
+  font-weight: bolder;
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:78]: TIL that `bolder` is a valid font weight.

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+a:not([href]):not([class]) {
+  color: inherit;
+  text-decoration: none;
+}
+
+a:not([href]):not([class]):hover {
+  color: inherit;
+  text-decoration: none;
+}
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:120]: This could use SASS nesting here, as is done in the original `_reboot.scss`.

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+  vertical-align: middle;
+}
+
+table {
+  border-collapse: collapse;
+}
+
+caption {
+  padding-top: 0.75rem;
+  padding-bottom: 0.75rem;
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:157]: There are variables for this in the upstream source. It would be helpful to use those here.

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+  text-align: left;
+  caption-side: bottom;
+}
+
+th {
+  text-align: inherit;
+  text-align: -webkit-match-parent;
+}
+
+h1,
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:168]: There are some form control styles upstream that are not present here. I guess we're omitting them because you can't insert those controls in page content?

File: lms/static/styles/moodle_pages/moodle_pages.scss [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+  margin-bottom: 0.5rem;
+  font-weight: 700;
+  line-height: 1.2;
+}
+
+h1 {
+  font-size: 2.34375rem;
+}
+
+@media (max-width: 1200px) {
```
Comment (robertknight) [inline lms/static/styles/moodle_pages/moodle_pages.scss:183]: Where do these media queries come from? I can't obviously see the matching code in the upstream sources.

**Review by robertknight — APPROVED** [review]
I had a query about the media queries for headings. Otherwise this looks good.

File: lms/static/styles/moodle_pages/moodle_pages.scss (hunk 6 above)
Comment (acelaya) [reply lms/static/styles/moodle_pages/moodle_pages.scss:78]: If you want an even bolder one, you can set `boldest` 😝

File: lms/static/styles/moodle_pages/moodle_pages.scss (hunk 9 above)
Comment (acelaya) [reply lms/static/styles/moodle_pages/moodle_pages.scss:168]: Yes, I intentionally skipped styles for elements that cannot be created by the WYSIWYG editor.

File: lms/static/styles/moodle_pages/moodle_pages.scss (hunk 8 above)
Comment (acelaya) [reply lms/static/styles/moodle_pages/moodle_pages.scss:157]: I have mixed feelings about using too many variables, or in other words, matching too much the non-processed source stylesheets.

I think that might make it harder to evolve in future, as it would make us check multiple source files (which are not very easy to navigate. The structure is relatively complex).

Considering the styles we need are simple, it's more straightforward to compare the actual output/generated CSS and pick the differences from there, if any.

In other words, I think the generated CSS is less likely to change than the source files, that might be refactored, variables renamed, etc, without changing produced CSS, making it harder to find the actual differences in future.

File: lms/static/styles/moodle_pages/moodle_pages.scss (hunk 10 above)
Comment (acelaya) [reply lms/static/styles/moodle_pages/moodle_pages.scss:183]: This is not set as is. They have a couple functions to generate media blocks, like this one https://github.com/moodle/moodle/blob/70f611116d055493e065edea6e82b853f02caf51/theme/boost/scss/bootstrap/mixins/_breakpoints.scss#L58

I don't know exactly how they end up generating this, as it's a bit hard to follow, but I also don't want to link to a lot of files because it's only going to make it harder to check.
